src/ingestors/rarible/index.ts

```diff
@@ -0,0 +1,112 @@
+    }   
+
+    async supportsContract(resources: MintIngestorResources, contract: MintContractOptions): Promise<boolean> {
+        const { chainId, contractAddress } = contract;
+        if (!chainId || !contractAddress) {
+          return false;
+        }
+    
+        const url = await urlForValidRaribleCollection(chainId, contractAddress);
+        return !!url;
+    }
+    
+
+    async createMintTemplateForUrl(resources: MintIngestorResources, url: string): Promise<MintTemplate> {
+        const isCompatible = await this.supportsUrl(resources, url);
+        if (!isCompatible) {
+          throw new MintIngestorError(MintIngestionErrorName.IncompatibleUrl, 'Incompatible URL');
+        }
+
+        const { chainId, contractAddress } = await raribleOnchainDataFromUrl(url);
+        if (!chainId || !contractAddress) {
+          throw new MintIngestorError(MintIngestionErrorName.MissingRequiredData, 'Missing required data');
+        }
+
+        return this.createMintForContract(resources, { chainId, contractAddress, url });
+    }
+
+    async createMintForContract(resources: MintIngestorResources, contract: MintContractOptions): Promise<MintTemplate> {
+        const { chainId, contractAddress } = contract;
+        if (!chainId || !contractAddress) {
+            throw new MintIngestorError(MintIngestionErrorName.MissingRequiredData, 'Missing required data');
+        }
+
+        const isValid = await urlForValidRaribleCollection(chainId, contractAddress)
+
+        if (!isValid) {
+          throw new MintIngestorError(MintIngestionErrorName.CouldNotResolveMint, 'Project not found');
+        }
+        const mintBuilder = new MintTemplateBuilder()
+            .setMintInstructionType(MintInstructionType.EVM_MINT)
+            .setPartnerName('Rarible');
+
+        if (contract.url) {
+            mintBuilder.setMarketingUrl(contract.url);
+        }
+        
+       const { 
+            name, 
+            description, 
+            imageURI, 
+            unitMintPrice, 
+            unitMintTokenAddress, 
+            unitPerWallet, 
+            conditionProof, 
+            ownerAddress,
+            startDate,
+            stopDate
+        } = await getRaribleContractMetadata(chainId, contractAddress, resources.alchemy)
+      
+        mintBuilder.setName(name).setDescription(description).setFeaturedImageUrl(imageURI);
+        mintBuilder.setMintOutputContract({ chainId: chainId, address: contractAddress });
+        mintBuilder.setCreator({
+                name: 'not available',
+                walletAddress: ownerAddress,
+        });
+
+        let totalPriceWei;
+        if (unitMintTokenAddress == '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE' || unitMintTokenAddress == '0x4200000000000000000000000000000000000006') {
+            totalPriceWei = unitMintPrice;
+        } else {
+            totalPriceWei = await getRaribleMintPriceInEth(unitMintPrice, unitMintTokenAddress);
```

Comment:
Is this common?

FWIW we don't currently support mints for mobile minting in non-native currencies (signer wallets have ETH on Ethereum network chains, and SOL on Solana)

Reply:
We could exclude non ether mints. I observed at the time of making this that there were a handful of non ether mints mostly in degen.